### PR TITLE
Bypass resource group when the proc is exiting in progress

### PR DESCRIPTION
**Pull request by xiong-gang**
Commit 4d5a653 fix the resource group wait queue corruption issue. On top of
that, this commit bypass resource group when the proc is exiting so that the
proc can terminate immediately.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/utils/resgroup/resgroup.c [hunk 1]
```diff
@@ -1927,15 +1927,9 @@ groupAcquireSlot(ResGroupInfo *pGroupInfo, bool isMoveQuery)
...
-	 *
-	 * Need to handle a special case, when MyProc was interrupted
-	 * by SIGTERM while waiting for resource group slot.
-	 * Some callbacks - RemoveTempRelationsCallback for example -
-	 * open new transactions on proc exit. It can cause a double
-	 * add of MyProc to the waiting queue (and its corruption).
 	 */
-	if (!procIsWaiting(MyProc))
-		groupWaitQueuePush(group, MyProc);
+	Assert(!procIsWaiting(MyProc) && !proc_exit_inprogress);
```
Comment (zhangh43) [inline src/backend/utils/resgroup/resgroup.c:1931]: In `ShouldAssignResGroupOnMaster `, either `!procIsWaiting(MyProc)` or `!proc_exit_inprogress` will bypass the resource group. But the assert here use `&&`. Let's keep them consistent.

Comment (darthunix) [reply src/backend/utils/resgroup/resgroup.c:1931]: I think that we don't need `Assert(!procIsWaiting(MyProc)` here as [groupWaitQueuePush() ](https://github.com/greenplum-db/gpdb/blob/c03061cf4daf12bc07df9eb13d92a83b702091ab/src/backend/utils/resgroup/resgroup.c#L3551) already has one.

Comment (xiong-gang) [reply src/backend/utils/resgroup/resgroup.c]: > In `ShouldAssignResGroupOnMaster `, either `!procIsWaiting(MyProc)` or `!proc_exit_inprogress` will bypass the resource group. But the assert here use `&&`. Let's keep them consistent.

either procIsWaiting(MyProc) or proc_exit_inprogress will bypass the resource group, in other words, it gets here only when `!procIsWaiting(MyProc)` and `!proc_exit_inprogress`

Comment (xiong-gang) [reply src/backend/utils/resgroup/resgroup.c]: > I think that we don't need `Assert(!procIsWaiting(MyProc)` here as [groupWaitQueuePush() ](https://github.com/greenplum-db/gpdb/blob/c03061cf4daf12bc07df9eb13d92a83b702091ab/src/backend/utils/resgroup/resgroup.c#L3551) already has one.

yes, removed it. Thanks

**Review by darthunix — APPROVED** [review]

**Pull request merged**